### PR TITLE
Run UpdateRanged only for active NPCs

### PR DESCRIPTION
**Pull request by Vordenburg**
Would run on dead NPCs previously.

### CODE REVIEW

File: Content.Server/NPC/Systems/NPCCombatSystem.Ranged.cs [hunk 1]
```diff
@@ -50,9 +50,9 @@ private void UpdateRanged(float frameTime)
...
         var xformQuery = GetEntityQuery<TransformComponent>();
         var combatQuery = GetEntityQuery<CombatModeComponent>();
-        var query = EntityQueryEnumerator<NPCRangedCombatComponent, TransformComponent>();
+        var query = EntityQueryEnumerator<NPCRangedCombatComponent, TransformComponent, ActiveNPCComponent>();
 
-        while (query.MoveNext(out var uid, out var comp, out var xform))
+        while (query.MoveNext(out var uid, out var comp, out var xform, out _))
         {
             if (comp.Status == CombatStatus.Unspecified)
                 continue;
```
Comment (metalgearsloth) [inline Content.Server/NPC/Systems/NPCCombatSystem.Ranged.cs:58]: Do you know what's not cleaning up? As the comp should be getting dumped when the plan is over.

Comment (Vordenburg) [reply Content.Server/NPC/Systems/NPCCombatSystem.Ranged.cs:58]: I've had a look into it more thoroughly. It appears at no time does UpdateRanged set a failing CombatStatus for the case of the NPC not being alive. This CombatStatus would be picked up by the RangedOperator and used to decide if the comp would be dumped, as mentioned.

For what it's worth, the Melee system works in this way, not setting a failing status when dead but ostensibly relying on ActiveNPCComponent.

https://github.com/space-wizards/space-station-14/blob/3fa99b8abf41463cf414c2931321e69467c73848/Content.Server/NPC/Systems/NPCCombatSystem.Melee.cs#L96-L107

Comment (metalgearsloth) [reply Content.Server/NPC/Systems/NPCCombatSystem.Ranged.cs:58]: I think I fixed this in the ranged NPC branch as there were a couple of issues. Ideally the mob dying should just dump their plan and in turn dump all of their operators / components.

Can you check if it's still an issue on nyano?

Comment (Vordenburg) [reply Content.Server/NPC/Systems/NPCCombatSystem.Ranged.cs:58]: Looks like the ranged NPC branch fixed the issue. I'll close this now. Thank you.

**Pull request closed without merging**